app/models/course/assessment/submission.rb

```diff
@@ -135,6 +138,16 @@ def publish(_ = nil)
     end
   end
 
+  # Handles the unsubmission of a submitted submission.
+  #
+  # This will revert all manually graded answers to an attempting state.
```

Comment:
I think we have to unsubmit all the latest answers. If you re-submit, you will get

```
Workflow::NoTransitionAllowed in Course::Assessment::Submission::SubmissionsController#update
There is no event finalise defined for the submitted state
```

Also, a student can't edit his code for auto_gradable coding questions (you get the interface where you can comment on the code) since they are in the submitted state.